lib/kinesis_client/stream/coordinator.ex

```diff
@@ -252,17 +261,23 @@ defmodule KinesisClient.Stream.Coordinator do
           )
       end
 
-    case result do
-      %{"StreamDescription" => %{"HasMoreShards" => true, "Shards" => shards}} ->
+    case describe_result do
+      {:ok, %{"StreamDescription" => %{"HasMoreShards" => true, "Shards" => shards}}} ->
         get_shards(
           stream_name,
           kinesis_opts,
           hd(Enum.reverse(shards))["ShardId"],
           shards ++ shard_list
         )
 
-      %{"StreamDescription" => %{"HasMoreShards" => false, "Shards" => shards} = stream} ->
-        stream |> Map.put("Shards", shards ++ shard_list)
+      {:ok, %{"StreamDescription" => %{"HasMoreShards" => false, "Shards" => shards} = stream}} ->
+        {:ok, Map.put(stream, "Shards", shards ++ shard_list)}
+
+      {:error, _} = error ->
+        error
```

Comment:
does this return the `error` rather than `{:error, error}`? is that inconsistent with the other clauses?

Reply:
The idea is that the response is already `{:error, any()}` so we just return it as is, rather than wrapping it another layer of tuple nesting.

Reply:
ohhh that's right. still learning to read these 😅